test/es-module/test-esm-child-process-fork-main.mjs

```diff
@@ -0,0 +1,19 @@
+import '../common/index.mjs';
+import assert from 'assert';
+import { fork } from 'child_process';
+import { once } from 'events';
+
+if (process.argv[2] !== 'child') {
+  const currentFile = 'test-esm-child-process-fork-main.mjs';
+  const { pathname: filename } = new URL(currentFile, import.meta.url);
+  const cp = fork(filename, ['child']);
+  const message = 'Hello World';
+  cp.send(message);
+
+  const [received] = await once(cp, 'message');
+  assert.deepStrictEqual(received, message);
+
+  cp.disconnect();
```

Comment:
To make sure we don't leave open process around:
```suggestion
  cp.disconnect();
  await once(cp, 'exit');
```

Reply:
fixed!